From b404f4b013ceea4f7bd4af912858da5cb3398e69 Mon Sep 17 00:00:00 2001
From: Martin Kosek <mkosek@redhat.com>
Date: Tue, 1 Apr 2014 16:23:14 +0200
Subject: [PATCH 114/116] Make ipa-client-automount backwards compatible

ipa-client-automount calls automountlocation-show command during the
process. Unfortunately, FreeIPA commands are forward compatible only
and thus fail the installer.

Similarly to ipa-client-install, call XML-RPC interface directly
with version fixed to 2.0 (command was already available at that
version) to fix the failure.

https://fedorahosted.org/freeipa/ticket/4290

Reviewed-By: Rob Crittenden <rcritten@redhat.com>
---
 ipa-client/ipa-install/ipa-client-automount | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/ipa-client/ipa-install/ipa-client-automount b/ipa-client/ipa-install/ipa-client-automount
index 8e0ba3d6c4499a1829caa335b248a75cdd581dc6..0031581652f7c18cfe25c2a7bc423185a9956168 100755
--- a/ipa-client/ipa-install/ipa-client-automount
+++ b/ipa-client/ipa-install/ipa-client-automount
@@ -433,7 +433,12 @@ def main():
         except errors.KerberosError, e:
             sys.exit('Cannot connect to the server due to ' + str(e))
         try:
-            api.Command['automountlocation_show'](unicode(options.location))
+            # Use the RPC directly so older servers are supported
+            result = api.Backend.xmlclient.forward(
+                'automountlocation_show',
+                unicode(options.location),
+                version=u'2.0',
+            )
         except errors.VersionError, e:
             sys.exit('This client is incompatible: ' + str(e))
         except errors.NotFound:
-- 
1.9.3

